### PR TITLE
Built new status labels

### PR DESCRIPTION
**Pull request by kwm14**
## Done
- Built a new set of `status` buttons
  - `p-button--validated`
  - `p-button--in-progress`
  - `p-button--new`
  - `p-button--updated`
  - `p-button--depreacated`
- Added   `@include vf-p-status;` to _vanilla.scss file
- Created seperate example pages for each button rather them all being in the same file

## QA

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/vanilla-framework/
- Check new buttons look like the desired design and are unclickable
  - http://0.0.0.0:8101/examples/patterns/status/validated/
  - http://0.0.0.0:8101/examples/patterns/status/in-progress/
  - http://0.0.0.0:8101/examples/patterns/status/new/
  - http://0.0.0.0:8101/examples/patterns/status/updated/
  - http://0.0.0.0:8101/examples/patterns/status/deprecated/
- Check documentation page
  - https://vanilla-framework-canonical-web-and-design-pr-2416.run.demo.haus/patterns/labels/

## Details
Fixes https://github.com/canonical-web-and-design/vanilla-framework/issues/2412

## Screenshots
Screenshot 2019-07-23 at 12 06 20
Screenshot 2019-07-23 at 12 06 09
Screenshot 2019-07-23 at 12 06 33
Screenshot 2019-07-23 at 12 06 56
Screenshot 2019-07-23 at 12 05 47

### CODE REVIEW

File: scss/_patterns_status.scss [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  font-size: 1rem;
+  line-height: 1.5rem;
+  padding: calc(#{$spv-nudge - $sp-unit * 0.5} - 1px) $sph-inner--small * 1.5;
+  text-align: center;
+  text-decoration: none;
+  display: inline-block;
+}
+
+// Override local variables
+@mixin vf-p-button-validated {
```
Comment (anthonydillon) [inline scss/_patterns_status.scss]: I think `vf-p-status-validated` is a better name as they are not buttons.

File: scss/_patterns_status.scss [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+// Default status button styling
+@mixin vf-p-status {
+  @include vf-p-button-validated;
+  @include vf-p-button-in-progress;
+  @include vf-p-button-new;
+  @include vf-p-button-updated;
+  @include vf-p-button-deprecated;
+}
+
+%p-button--status {
```
Comment (anthonydillon) [inline scss/_patterns_status.scss]: This should just be `%vf-status`

**Comment by kwm14** [comment]
> I prefer the to make this a new pattern then piggyback off the buttons pattern. Also, this PR is missing documentation.

Sure thing I can update. In regards to docuemntation, I have a separate [issue](https://github.com/canonical-web-and-design/vanilla-framework/issues/2413) for this, unless you want me to add it to this PR?

**Comment by anthonydillon** [comment]
It is much better to do the sass, examples and docs in the same PR so that it can all be reviewed at the same time. That is the main reason for having the docs in the codebase.

File: scss/_patterns_status.scss [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  text-align: center;
+  text-decoration: none;
+  display: inline-block;
+}
+
+// Override local variables
+@mixin vf-p-button-validated {
+  .p-button--validated {
+    @extend %p-button--status;
+    background-color: #006b75;
```
Comment (anthonydillon) [inline scss/_patterns_status.scss]: You need set hex values to a variable before using. I would suggest setting the vars at the top of the main mixin (vf-p-status) so its easy to change in the future.

**Comment by spencerbygraves** [comment]
@kwm14 @anthonydillon should we be calling these buttons?

**Comment by deadlight** [comment]
Can we rename these "p-status--"?

**Comment by kwm14** [comment]
> This is looking much better but missing documentation.

I know I know. I'm not that fast 😆

**Comment by kwm14** [comment]
> Can we rename these "p-status--"?

Sure thing. Conversation with @spencerbygraves we've agreed that `'label` will be better than `status` as these are used in JAAS as tags.

JAAS - Tag labels
Vanilla - Status labels

**Comment by anthonydillon** [comment]
Just missing some docs :-1:

**Comment by kwm14** [comment]
## Documentation
- Added docs page to list out labels
- Added link to new component status page (will build next) so will 404
- Added link to labels design spec (will add to Design repo) so will 404

## QA

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/patterns/labels/
- Review page 
- Make sure label examples are displayed

**Comment by kwm14** [comment]
@anthonydillon added documentation page ❤️

File: docs/patterns/labels.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
+---
+layout: default
+---
+
+## Labels
+
+<hr>
+
+Labels are static elements which you can apply to HTML elements to signify status, tags or any other information you find useful.
```
Comment (anthonydillon) [inline docs/patterns/labels.md]: I don't think you need this part of the sentence "to HTML elements"

File: docs/patterns/labels.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+View example of the labels pattern
+</a>
+
+### In progress
+
+Label to be used when a design spec and code implementation are not yet finished.
+
+<a href="/examples/patterns/labels/in-progress/"
+    class="js-example">
+View example of the labels pattern
```
Comment (anthonydillon) [inline docs/patterns/labels.md]: Can the change to the name of the pattern for a11y

File: docs/patterns/labels.md (hunk 4 above)
Comment (kwm14) [reply docs/patterns/labels.md]: Fixed 👍

File: docs/patterns/labels.md (hunk 5 above)
Comment (kwm14) [reply docs/patterns/labels.md]: Fixed 👍

**Comment by kwm14** [comment]
@spencerbygraves ready for design review 👍

**Pull request merged**